pkg/services/pluginsintegration/plugincontext/plugincontext.go

```diff
@@ -79,6 +80,11 @@ func (p *Provider) Get(ctx context.Context, pluginID string, user identity.Reque
 		if err != nil {
 			return backend.PluginContext{}, err
 		}
+
+		if plugin.APIVersion != "" && appSettings.APIVersion != "" && plugin.APIVersion != appSettings.APIVersion {
+			return backend.PluginContext{}, fmt.Errorf("plugin API version mismatch, plugin: %s, datasource: %s", plugin.APIVersion, appSettings.APIVersion)
```

Comment:
🤔 I don't think this is an error.  IIUC the plugincontext will pass along the api_version whatever it is.  The plugin itself needs to know if it can support it or not.

At this stage (with either 0 or 1 apiVersions) an error is OK, but i don't think that makes sense

Reply:
I think the only place we can error now is when saving the config with an apiVersion != to the one saved in plugin.json

Reply:
I guess it depends on how early we want to fail. If we don't error here, the plugin will receive a request in which the pluginContext and the datasource / app settings won't match. A plugin may decide to support multiple apiVersions through migrations so fair point to avoid this error here.

> I think the only place we can error now is when saving the config with an apiVersion != to the one saved in plugin.json

How I coded this at the moment is that the apiVersion is not part of the config save request (and it's generated at the time of saving, with the version of the plugin running in the server). Probably, it's better to just rely on the request value but I don't think we need to error in that case either: A server may be running an older version from the one in the request and if this server is just saving the config (not using it), it's okay to ignore the version mismatch.